### DISCUSSION TITLE
Frustration as a new Flame developer

### DISCUSSION THREAD

**Discussion by cherry-vanilla** [flame-engine/flame#1401] [General]
Hi team,

Congrats on shipping v1.0! A huge milestone for any project. I can appreciate the hard work that went into this.

Reason I'm posting is that I wanted to offer the perspective of someone new to Flame. I'm a programmer and I have an idea for game I want to make, but I'm new to game development. I reviewed a ton of game engines and narrowed it down to deciding between Unity & Flame. Due to my desire to learn more about Flutter, I dove into Flame for about 12 hours over 3 days. It's not been very smooth unfortunately.

The [tutorial link](https://tutorials.flame-engine.org/#/) from the flame homepage unfortunately doesn't have much on it. And the [Awesome flame](https://github.com/flame-engine/awesome-flame#articles--tutorials) repo has lots to learn from, but after spending hours trying these, it turns out basically none of them work, due to big changes in Flutter and Flame since they were published. Google and YouTube searches don't have a lot more to offer, and most of the tutorials they point to are broken.

It would be extremely helpful (and user-friendly) if you could update the Awesome Flame page to show which tutorials and guides are currently functional. This would be incredibly helpful and grow the Flame community.

Thanks!

**Comment by st-pasha** [comment]
Thanks Cherry for your feedback.
You are of course right that there are many tutorials out there (including on Awesome-Flame) that were written for an older version of Flame and therefore are difficult to follow.

We do have current plans to expand the tutorial landscape. For example two of the current PRs (#1376 and #1393) are aimed at tutorials; and we will be adding more in the near future.

**Comment by cherry-vanilla** [comment]
That's great to hear. I'll stay tuned.

**Comment by spydon** [comment]
You are completely right ofc, a remedy meanwhile can be to check out community made tutorials.

**YouTube**:
 - Ryuzaki's Dino run tutorial:
https://www.youtube.com/playlist?list=PLiZZKL9HLmWOmQgYxWHuOHOWsUUlhCCOY

**Article tutorials**:
 - Building Games in Flutter with Flame: Getting Started - https://www.raywenderlich.com/27407121-building-games-in-flutter-with-flame-getting-started

 - Devowl's Flutter & Flame series: 
 - Step 1: Create your game - https://medium.com/flutter-community/flutter-flame-step-1-create-your-game-b3b6ee387d77  
 - Step 2: Game basics - https://medium.com/flutter-community/flutter-flame-step-2-game-basics-48b4493424f3
 - Step 3: Sprites and inputs - https://blog.devowl.de/flutter-flame-step-3-sprites-and-inputs-7ca9cc7c8b91
 - Step 4: Collisions & Viewport - https://blog.devowl.de/flutter-flame-step-4-collisions-viewport-ff2da048e3a6
 - Step 5: Level generation & camera - https://blog.devowl.de/flutter-flame-step-5-level-generation-camera-62a060a286e3
 - Step 6: Effects and sounds - https://blog.devowl.de/flutter-flame-step-6-effects-and-sounds-f65a1d30189b
 - Step 7: Play the game - https://blog.devowl.de//flutter-flame-step-7-play-the-game-1eb4f0dc0299
 - Step 8: Effects and ads - https://blog.devowl.de/flutter-flame-effects-and-ads-3e243009d18c
 
 Devowl's series is the one most up to date.

**Reply by st-pasha** [reply]
Perhaps we could merge the Awesome-Flame list into the tutorials section on the Docs site, so that it would be both easier to find, and easier to curate for freshness. WDYT?

**Reply by spydon** [reply]
That could be a good idea, that is how Melos have done it too. Wdyt @flame-engine/flame-colab?

**Reply by erickzanardo** [reply]
By merging you mean removing the awesome repository?

If so, I am not sure, awesome lists have this standards that it needs to follow so it can be listed on the main awesome list, which Flame is right now and I would not like to loose that.

**Reply by spydon** [reply]
Maybe we should simply properly link and explain it then?

**Reply by st-pasha** [reply]
Since awesome-flame is just a single markdown document, we could set up the docs site so that it copies that document during the building, and then the sphinx would integrate it with the rest of the documentation.

**Reply by erickzanardo** [reply]
> Since awesome-flame is just a single markdown document, we could set up the docs site so that it copies that document during the building, and then the sphinx would integrate it with the rest of the documentation.

This I like!

**Reply by cherry-vanilla** [reply]
Thanks spydon. Great to hear the team is putting thought into the tutorials. I tried these tutorials previously. I know your team didn't make these, but here's some comments on them anyway, if it helps inform you linking to them from your site.

DinoRun: doesn't work as-is due to Flutter & Flame updates, and as a first exposure to Flame it was frustrating. The last video in the series tries to fix the issues with the first 8 videos. But that's like taking a math class, and the professor says all the equations in the textbook are wrong, so you need to read the appendix for each equation to see if it's accurate or not. It's not great for learning :)

raywenderlich: example was promising at first when I looked at it a couple days ago. And I applaud the author for making it. The problem with that one is that he skims over a lot of stuff, and doesn't explain all that much in detail. It assumes the reader already knows Flutter & Dart pretty well.

DevOwl: Making my way through this one currently.

In an ideal world, JapAlekhin's [tutorial](https://jap.alekhin.io/2d-casual-mobile-game-tutorial-flame-flutter-part-1) could be re-written so it works properly (either by him or someone else) as that is by far the best subject matter on learning Flame I could find.

**Reply by spydon** [reply]
Thanks for the feedback, it is very valuable since it is hard for us to put ourselves in the shoes of someone new to flame. :sweat_smile: 
We'll take this into consideration when making the new tutorials!

**Comment by cherry-vanilla** [comment]
FYI: This just popped up in a Google search for me. Posted yesterday. Haven't tried it yet.

https://blog.devgenius.io/lets-create-a-snake-game-using-flutter-and-flame-38482d3cf0ff

**Comment by ufrshubham** [comment]
Just saw this discussion and thought that maybe I should also mention some frustrations as someone creating content around Flame 😅. The biggest problem I face is making people, new to Flame, understand how package version and breaking changes work. Almost 90% of the time they want to use the latest version of Flame while following tutorials made using older versions. It is very hard to convince them that it is not end of the world if they stick to an older version. Also, unlike other engines (or software in general) Flame doesn't have something like a LTS version, that I can point to and confidently say that it will stay free of breaking changes for a relatively longer duration of time. I did try asking viewers to stick to same version of packages as shown in the videos, but everyone seems to love that `^` sign before their package version😄.

I wish YouTube had some way to update existing videos. But then remaking Dino Run every time a new version of Flame drops doesn't sound all that exciting. It feels like written form of tutorials are the best because I can at least keep updating them as Flame evolves.
